pydoctor/templatewriter/__init__.py

```diff
@@ -5,6 +5,364 @@
+
+            if path.suffix.lower() == '.html':
+                return _HtmlTemplate(path=path)
+            else:
+                return _StaticTemplate(path=path)
+        else:
+            warnings.warn(f"Cannot create Template: {path.as_posix()} is not a template file.")
+        return None
+    
+    def is_empty(self) -> bool:
+        """
+        Does this template is empty? 
+        Empty placeholders templates will not be rendered. 
+        """
+        return len(self.text.strip()) == 0
+
+    @property
+    def name(self) -> str:
+        """
+        Template filename
+        """
+        return self.path.name
+
+    @abc.abstractproperty
+    def version(self) -> int:
+        """
+        Template version, C{-1} if no version. 
+
+        HTML Templates should have a version identifier as follow::
+    
+            <meta name="pydoctor-template-version" content="1" />
+
+        This is always C{-1} for CSS and JS templates. 
+        """
+        raise NotImplementedError()
+
+    @abc.abstractproperty
+    def loader(self) -> Optional[ITemplateLoader]:
+        """
+        Object used to render the final file. 
+
+        For HTML templates, this is a L{ITemplateLoader} or a list of L{ITemplateLoader}.  
+
+        For CSS and JS templates, this is C{None} 
+        because there is no rendering to do, it's already the final file.  
+        """
+        raise NotImplementedError()
+
+class _StaticTemplate(Template):
+    """
+    Static template: no rendering. 
+    For CSS and JS templates. 
+    """
+    @property
+    def version(self) -> int: return -1
+    @property
+    def loader(self) -> None: return None
+
+
+@implementer(ITemplateLoader)
+class _NullLoader:
+    def load(self) -> List[str]:
+        return []
+
+class _HtmlTemplate(Template):
+    """
+    HTML template that works with the Twisted templating system 
+    and use L{xml.dom.minidom} to parse the C{pydoctor-template-version} meta tag. 
+    """
+    def __init__(self, path: Path):
+        super().__init__(path)
+        self._version: int
+        self._loader: ITemplateLoader
+        if self.is_empty():
+            self._version = -1
+            self._loader = _NullLoader()
+        else:
+            self._version = self._extract_version(parse_dom(self.text), self.name)
+            self._loader = XMLString(self.text)
+    
+    @property
+    def version(self) -> int: return self._version
+    @property
+    def loader(self) -> ITemplateLoader: return self._loader
+
+    @staticmethod
+    def _extract_version(dom: minidom.Document, template_name: str) -> int:
+        # If no meta pydoctor-template-version tag found, 
+        # it's most probably a placeholder template. 
+        version = -1
+        for meta in dom.getElementsByTagName("meta"):
+            if meta.getAttribute("name") != "pydoctor-template-version":
+                continue
+            if meta.hasAttribute("content"):
```

Comment:
We could handle the absence first:
```py
if not meta.hasAttribute("content"):
    warnings.warn(...)
    continue
```
I think that would help readability, since the check and the warning message are closer together then.